library/src/interactor_impl.cxx

```diff
@@ -172,228 +176,142 @@ class interactor_impl::internals
       return;
     }
 
-    // No user defined behavior, use standard behavior
-    vtkRenderWindow* renWin = self->Window.GetRenderWindow();
-    vtkF3DRenderer* ren = vtkF3DRenderer::SafeDownCast(renWin->GetRenderers()->GetFirstRenderer());
-    bool checkColoring = false;
-    bool printColoring = false;
-    bool render = false;
-
     // Available keycodes: None
     switch (keyCode)
     {
       case 'W':
-        self->AnimationManager->CycleAnimation();
-        self->Options.scene.animation.index = self->AnimationManager->GetAnimationIndex();
-        ren->SetAnimationnameInfo(self->AnimationManager->GetAnimationName());
-        render = true;
+        self->Interactor.triggerCommand("cycle_animation");
```

Comment:
same remark as for the app. A map instead of the switch/case would look better

Reply:
same answer